File: src/main/java/io/bindernews/thegrackle/cards/Takeoff.java
package io.bindernews.thegrackle.cards;

import com.megacrit.cardcrawl.actions.common.GainBlockAction;
import com.megacrit.cardcrawl.core.AbstractCreature;
import io.bindernews.bnsts.CardNums;
import io.bindernews.thegrackle.stance.StanceAloft;

public class Takeoff extends BaseCard {
    public static final CardConfig CFG = new CardConfig("Takeoff", CardType.SKILL);
    public static final CardNums NUM = CardNums.builder().cost(1).block(6).blockUpg(10).build();

    public Takeoff() {
        super(CFG, CardRarity.BASIC, CardTarget.SELF);
        tags.add(CardTags.STARTER_DEFEND);
        NUM.init(this);
    }

    @Override
    public void apply(AbstractCreature p, AbstractCreature m) {
        addToBot(new GainBlockAction(p, p, block));
        addToBot(iop().changeStance(p, StanceAloft.STANCE_ID));
    }

    @Override
    public void upgrade() {
        NUM.upgrade(this, false);
    }
}

File: src/main/kotlin/io/bindernews/thegrackle/cards/Takeoff.kt
package io.bindernews.thegrackle.cards

import com.megacrit.cardcrawl.actions.common.GainBlockAction
import com.megacrit.cardcrawl.core.AbstractCreature
import io.bindernews.thegrackle.helper.CardVariables
import io.bindernews.thegrackle.helper.ModInterop
import io.bindernews.thegrackle.stance.StanceAloft

class Takeoff : BaseCard(C, VARS) {
    override fun apply(p: AbstractCreature, m: AbstractCreature?) {
        addToBot(GainBlockAction(p, p, block))
        addToBot(ModInterop.iop().changeStance(p, StanceAloft.STANCE_ID))
    }

    companion object {
        @JvmField
        val C = CardConfig("Takeoff", CardType.SKILL, CardRarity.BASIC, CardTarget.SELF)
        val VARS = CardVariables.config { c ->
            c.cost(1, -1)
            c.block(6, 10)
            c.tags(CardTags.STARTER_DEFEND)
        }
    }
}